### ISSUE TITLE
The zero-shot classification example on the ChestX-ray 14 dataset does not produce reasonable results.

### ISSUE THREAD

**Issue by yangzhou12** [MediaBrain-SJTU/MedKLIP#1]
Thanks for the interesting work. 
I am trying to run the zero-shot classification example on the ChestX-ray 14 dataset by using the provided checkpoint.
Since "test.csv" is not provided, I slightly modified your code and use "BBox_List_2017_Official_NIH.csv" in the original ChestX-ray 14 dataset as the test file.

However, the model only produces unreasonable predictions such as 
tensor([0.9732, 0.8431, 0.2765, 0.1839, 0.7249, 0.8058, 0.2045, 0.0230, 0.0410, 0.0181, 0.9748, 0.9703, 0.9796, 0.9785]),
where most classes have large scores.

May I know if I am doing anything wrong in running the example?
Thank you very much!

**Comment by chaoyi-wu** [comment]
Thank you for your attention. You may calculate the AUC scores to check whether the prediction is correct.  The network may give a higher probability in absolute values for some commonly appearing diseases in the training set. So, AUC scores can better measure the correctness of results. We also test the model on the official set and the avgAUC is around 0.73. You may compare with yours to see whether the results are right.

**Comment by yangzhou12** [comment]
Thanks for the prompt reply.
Since all the images in "BBox_List_2017_Official_NIH.csv"are from the first 8 classes, I treat the test setting as the 8-class classification problem and get an avgAUC of 69.84.
As there exist multiple classes with similar scores, should we still take the class with the highest score as the final classification prediction (which may not be correct) or there is another way to get a better prediction?

**Comment by chaoyi-wu** [comment]
Sorry, I am a little confused about "the highest score as the final classification prediction". The CXR-14(or 8) is a multi-label dataset, not a multi-class dataset. The original 14-prediction scores are independent to each other, i.e, we calculate the final logits with sigmoid not softmax. So, you may choose a binary threshold for each class to determine the final14 (8) binary class binary predictions. We choose the threshold to get the maxF1 scores as claimed in the paper.  I'm not sure if this solves your confusion, you could email me directly if the problem details are complex, and if convenient, Chinese and English are all ok.

**Comment by yangzhou12** [comment]
Thanks for the explanations. I understand that the MedKLIP model is trained and tested in a multi-label manner. I am asking the question about multi-class classification just to make sure I use the MedKLIP model correctly, as each image in "BBox_List_2017_Official_NIH.csv" only has one label, leading to a multi-class classification problem.  
I have no further question and thank you again for the kind reply.
